File: Retos/Reto #35 - PRIMEROS PASOS [Fácil]/python/KevinED11.py

# Variables
name: str = "kevin"
age: int = 22
money_in_the_bank: float = 1.0
married: bool = True

# Constant
PI_VALUE = 3.1416

# Conditionals
if age >= 18:
    print("eres mayor de edad")
else:
    print("eres menor de edad")

if married:
    print("esta casado")
elif not married:
    print("no esta casado")
    
# Structures
names: list = ["kevin", "damian"]
person: dict =  {"name": "kevin", "age": 22}
unique_values: set = {"kevined11", "dani123", "kevined11"}
person_values: tuple = (person.get("name"), person.get("age")) 


# Loops
for value in person_values:
    print(value)

counter = 1
while counter <= 10:
    print(counter)
    counter += 1

# functions
def print_names(names: list) -> None:
    for name in names:
        print(names)

def hello():
    print("hello world")
  
def name1() -> str:
    return "kevin"


# Class
class Person:
    def __init__(self, name, age):
        self.nam = name
        self.age = age
      
# Exceptions
def is_person(person: dict) -> dict:
    if not isinstance(person, dict):
        raise TypeError("solo debes pasar diccionarios")
    
    return person


try:
    is_person(person)
except TypeError as err:
    print(err)
    

File: Retos/Reto #35 - PRIMEROS PASOS [Fácil]/elixir/KevinED11.exs
# Hello World
IO.puts("Hello world from Elixir")

# Variables
name = "kevin"
married = true
age = 22
money = 1.0

# Constants
IO.puts("It doesn't have traditional constants.")

# Structures
list = [1, 2, 3]
IO.puts(list)
tuple = {"hello", "world"}
map = %{:a => 1, 2 => :b}


# Functions
sum = fn (a, b) -> a + b end

def function_name(arg1, arg2) do
  # Code
end

# Loops
for(i = 0; i < 10; i++) {
   IO.puts(i);
}

# Conditionals
if age >= 18 do
  IO.puts("Eres mayor de edad")
else
  IO.puts("Eres menor de edad")
end

if condition1 do
  # código para ejecutar si condition1 es verdadera
elsif condition2 do
  # código para ejecutar si condition2 es verdadera
else
  # código para ejecutar si ninguna de las condiciones anteriores es verdadera
end

cond do
  age >= 18 ->
    IO.puts("Eres mayor de edad")
  age < 18 ->
    IO.puts("Eres menor de edad")
  true ->
    # default case


# Class
IO.puts("Elixir is a functional programming language in which classes as such do not exist, but modules are used instead. Modules serve as containers to group related functions.")


# Exeptions
def divide(a, b) do
  if b === 0 do
    {:error, "Divisor no puede ser cero"}
  else
    {:ok, a / b}
  end
end
